src/components/select/multi-select/multi-select.component.js

```diff
@@ -349,9 +357,22 @@ const MultiSelect = React.forwardRef(
           return valueList;
         });
       },
+      // eslint-disable-next-line react-hooks/exhaustive-deps
       [createCustomEvent, onChange, textboxRef]
     );
 
+    function isExpectedValue(singleValue, expectedValue) {
```

Comment:
See other comment, I don't think we need to have this duplicate code if you filter `(type === Option)` and pass `child.props.value` instead of the `child`

Reply:
moved to a separate file